src/GafferSceneModule/PrimitivesBinding.cpp

```diff
@@ -92,6 +93,14 @@ class LightSerialiser : public GafferBindings::NodeSerialiser
 		{
 			shaderNamePlug = light->getChild<Gaffer::StringPlug>( "__model" );
 		}
+		if( !shaderNamePlug )
+		{
+			// TODO: Should this be using shader plug instead?
```

Comment:
Are you referring to `__shaderIn`? If so, then no. Evaluating that would generate a shader network, which is much more expensive than necessary when we only want the shader name. Once we've updated the base class to force all derived classes to use internal shader nodes, we should be able to ditch all those other forms of getting a shader name plug, and just use this one.

Reply:
Removed todo in https://github.com/GafferHQ/gaffer/pull/3491/commits/ebdae729f2bd3487348dbca4aa22af7535fa236f